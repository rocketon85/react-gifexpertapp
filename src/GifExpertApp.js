import React, {useState} from "react";
import PropTypes from 'prop-types';
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
//PdVA4yBUM2y4hsugeWvkdDNCv40jOVKC

export const GifExpertApp = ({defaultCategories=[]}) => {
    const [categories, setCategories] = useState(defaultCategories);
    
    const handlerAdd = (cat) =>{
        let added = false;

        if(cat.trim().length > 2){
            setCategories([cat, ...categories]);
            added=true;
        }
        return added;
    }

    return  (
        <>
            <h2>GifExpertApp</h2>
            <hr />
            <AddCategory setCategories={handlerAdd}></AddCategory>

           <ol>
               {
                    categories.map( (category,idx) => (
                         <GifGrid 
                            key={category}
                            cat={category}
                        />
                    ))

               }
           </ol>
        </>
    )
}


export default GifExpertApp;