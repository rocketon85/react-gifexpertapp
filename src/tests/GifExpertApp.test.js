import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifExpertApp } from '../GifExpertApp';
// import { useFetchGifs } from '../../hooks/useFetchGifs';
// jest.mock('../../hooks/useFetchGifs');

describe('first', () => {
    test('should match snapshot', () => {
        const wrapper = shallow( <GifExpertApp  /> );

        expect(wrapper).toMatchSnapshot();
        
    });
  
    test('should must render two categories', () => {
        const cats = ['Dragon Ball', 'One Punch'];
        const wrapper = shallow( <GifExpertApp defaultCategories={cats} /> );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(cats.length);
        
    });
});
