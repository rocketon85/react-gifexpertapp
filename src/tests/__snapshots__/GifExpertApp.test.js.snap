// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first should match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`first should must render two categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      cat="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      cat="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
